### PR TITLE
rename(std/testing): rename assert*Contains to assert*Includes

### PR DESCRIPTION
**Pull request by timreichen**
#6888

# Rename functions in asserts.ts
- assertStringContains -> assertStringIncludes
- assertArrayContains -> assertArrayIncludes

### CODE REVIEW

**Comment by bartlomieju** [comment]
I'm generally in favor of this change, the only point of discussion is original suggestion from #6888 if we should have both `assertStringIncludes` and `assertArrayIncludes` or just `assertIncludes` as implemented in this PR. Given that `assertEquals` is magical and works on all values makes `assertIncludes` favorable. 

CC @kitsonk @nayeemrmn @caspervonb

**Comment by caspervonb** [comment]
Difficult to see what has actually changed here with 232 changes files but in favour of generalising on `a.includes(b)` as long as we provide rich pretty printed diffs for both strings and array-likes (otherwise `assert(a.includes(b))` is just as good.

File: std/testing/asserts.ts [hunk 1]
```diff
@@ -330,69 +331,66 @@ export function assertNotStrictEquals(
...
   msg?: string,
 ): void {
-  const missing: unknown[] = [];
-  for (let i = 0; i < expected.length; i++) {
-    let found = false;
-    for (let j = 0; j < actual.length; j++) {
-      if (equal(expected[i], actual[j])) {
-        found = true;
-        break;
+  if (typeof actual === "string" && typeof expected === "string") {
```
Comment (caspervonb) [inline std/testing/asserts.ts:363]: Surprising that you're special casing here, behavior should be the same across all types.

**Comment by nayeemrmn** [comment]
> if we should have both `assertStringIncludes` and `assertArrayIncludes` or just `assertIncludes` as implemented in this PR

I think they should definitely be different functions. They have different and unrelated behaviour, which would be especially confusing since strings are iterable. The JS methods are namespaced to `Array.prototype.includes()` and `String.prototype.includes()`. Making them overloads is illogical and makes both the implementation and call sites harder to read.

**Comment by timreichen** [comment]
@caspervonb The special casing is because of the overload. To avoid that, @nayeemrmn suggestion might be the better solution. Just a rename of `contains` to `includes`.

**Comment by bartlomieju** [comment]
> > if we should have both `assertStringIncludes` and `assertArrayIncludes` or just `assertIncludes` as implemented in this PR
> 
> I think they should definitely be different functions. They have different and unrelated behaviour, which would be especially confusing since strings are iterable. The JS methods are namespaced to `Array.prototype.includes()` and `String.prototype.includes()`. Making them overloads is illogical and makes both the implementation and call sites harder to read.

Thanks for being voice of reason @nayeemrmn. In this light I agree that we should just do the rename without merging those assertion. @timreichen could you please update the PR?

**Comment by timreichen** [comment]
@bartlomieju already did:)

File: Releases.md [hunk 2]
```diff
@@ -487,7 +487,7 @@ Changes in std version 0.59.0:
 - fix(std/node): fix readFile types, add encoding types (#6451)
 - fix(std/node): global process should usable (#6392)
 - fix(std/node/process): env, argv exports (#6455)
-- fix(std/testing) assertArrayContains should work with any array-like (#6402)
+- fix(std/testing) assertArrayIncludes should work with any array-like (#6402)
```
Comment (bartlomieju) [inline Releases.md:490]: There should be no changes to this file, revert

**Review by bartlomieju — CHANGES_REQUESTED** [review]

File: Releases.md (hunk 2 above)
Comment (timreichen) [reply Releases.md]: O shoot, I thought I excluded that file. Reverted.

**Comment by bartlomieju** [comment]
@timreichen this PR shouldn't change 3k lines diff - I think something went wrong during rebase, please check

**Comment by timreichen** [comment]
@bartlomieju Shoot, I suck at rebasing. Sorry. Done now.

**Review by bartlomieju — COMMENTED** [review]
LGTM and I'm in favor of landing this PR

@lucacasonato @kitsonk please take a look

**Comment by lucacasonato** [comment]
LGTM. We should definitely mention this in the blog post.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**